File: advent_of_code.py
from sys import argv

# from day_N_TASK_NAME import day_N_task_1, day_N_task_2
from day_1_trebuchet import day_1_task_1, day_1_task_2


def print_help():
	print('advent_of_code.py [-d day] [-t task]')
	print('    -d day  - day has values 1-25')
	print('    -t task - task has values 1 or 2')
	print('For other info check: https://adventofcode.com/')


if __name__ == '__main__':
	if '-d' in argv:
		arg_i = argv.index('-d')
		if arg_i + 1 > len(argv) - 1 or int(argv[arg_i + 1]) < 1 or int(argv[arg_i + 1]) > 25:
			print_help()
			exit(1)

		days = [int(argv[arg_i + 1])]
	else:
		days = [i for i in range(1, 26)]

	if '-t' in argv:
		arg_i = argv.index('-t')
		if arg_i + 1 > len(argv) - 1 or not int(argv[arg_i + 1]) in [1, 2]:
			print_help()
			exit(2)

		tasks = [int(argv[arg_i + 1])]
	else:
		tasks = [1, 2]

	for d in days:
		for t in tasks:
			try:
				print('Day', d, 'task', t, '- ', end='')
				locals()[f'day_{d}_task_{t}']()
			except:
				print('Solution does not exist!')
				exit(3)

File: advent_of_code.go
package main

import (
	"fmt"
	"os"
	"strconv"

	"github.com/isinek/aoc/days"
)

func printHelp() {
	fmt.Println("go run advent_of_code.go [-d day] [-t task]")
	fmt.Println("    -d day  - day has values 1-12")
	fmt.Println("    -t task - task has values 1 or 2")
	fmt.Println("For other info check: https://adventofcode.com/")
}

func main() {
	args := os.Args[1:]

	for _, v := range args {
		if v == "-h" {
			printHelp()
			os.Exit(0)
		}
	}

	var runDays []int
	var runTasks []int

	for i, v := range args {
		switch v {
		case "-d":
			if i+1 > len(args)-1 {
				printHelp()
				os.Exit(1)
			}

			day, err := strconv.Atoi(args[i+1])
			if err != nil || day < 1 || day > 12 {
				printHelp()
				os.Exit(1)
			}

			runDays = append(runDays, day)
		case "-t":
			if i+1 > len(args)-1 {
				printHelp()
				os.Exit(1)
			}

			task, err := strconv.Atoi(args[i+1])
			if err != nil || task < 1 || task > 2 {
				printHelp()
				os.Exit(1)
			}

			runTasks = append(runTasks, task)
		}
	}
	if len(runDays) == 0 {
		for i := 1; i <= 12; i++ {
			runDays = append(runDays, i)
		}
	}
	if len(runTasks) == 0 {
		runTasks = []int{1, 2}
	}

	for _, d := range runDays {
		var dayMap map[int]func()
		var ok bool

		if dayMap, ok = days.Solutions[d]; !ok {
			fmt.Printf("Solutions for day %d do not exist\n", d)
			os.Exit(0)
		}

		for _, t := range runTasks {
			fmt.Printf("Day %d task %d - ", d, t)

			if fn, ok := dayMap[t]; ok {
				fn()
				continue
			}

			fmt.Println("Solution does not exist!")
			os.Exit(2)
		}
	}
}
